### ISSUE TITLE
[tune] JupyterNotebookReporter not working with ray.util.connect()

### ISSUE THREAD

**Issue by al-rigazzi** [ray-project/ray#16016]
### What is the problem?
When using `ray.util.connect()` to run a `tune` workload in a Jupyter Notebook, the `JupyterNotebookReporter` does not show the correct output (i.e. the well-formatted table of trials). Instead, it only prints `<IPython.core.display.HTML object>`. Notice that when the cluster is started in the cell (thus using `ray.init()`), the behavior is as expected.

*Ray version and other system information (Python version, TensorFlow version, OS):*
- Ray 1.3.0
- Python 3.7.0
- OS: RHEL

### Reproduction (REQUIRED)
This code has to be run in a Jupyter Notebook cell, while the ray cluster is running:

```python
from ray.tune.progress_reporter import JupyterNotebookReporter
import ray
from ray import tune
import ray.util

ray.util.connect(<HEAD_HOST_ADDRESS> +":10001")

tune.run(
    "PPO",
    stop={"episode_reward_max": 200},
    config={
        "framework": "torch",
        "env": "CartPole-v0",
        "lr": tune.grid_search([0.001, 0.002, 0.003, 0.004, 0.005]),
    },
    progress_reporter = JupyterNotebookReporter(True)
)
```

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by mehes-kth** [comment]
The exact same problem occurs with Ray 1.6.0 using `ray.init(address=f'ray://{head}:10001')`.

**Comment by amogkam** [comment]
@mehes-kth yep, a fix for this is in progress! Will update here once we merge it.
